File: backend/backup/service/AuthService.java
package com.bankingsystem.service;

import com.bankingsystem.dto.*;

public interface AuthService {
    ApiResponse<?> register(RegisterRequest request);
    LoginResponse login(LoginRequest request);
    ApiResponse<?> verifyOtp(OtpRequest request);
    ApiResponse<?> resendOtp(String email);
    ApiResponse<?> forgotPassword(String email);
    ApiResponse<?> resetPassword(PasswordResetRequest request);
    UserResponse getCurrentUser(String email);
    UserResponse updateProfile(String email, UserResponse request);
}

File: frontend/src/services/AuthService.js
import api from './api'

const AuthService = {
  register: (data) => api.post('/api/auth/register', data),
  login: (data) => api.post('/api/auth/login', data),
  verifyOtp: (data) => api.post('/api/auth/verify-otp', data),
  resendOtp: (email) => api.post('/api/auth/resend-otp', { email }),
  forgotPassword: (email) => api.post('/api/auth/forgot-password', { email }),
  resetPassword: (data) => api.post('/api/auth/reset-password', data),
  getProfile: () => api.get('/api/auth/me'),
  updateProfile: (data) => api.put('/api/auth/update-profile', data),
}

export default AuthService
